### PR TITLE
Add a fuzzing preset

### LINKED ISSUE #210: Add a preset for fuzzing from the IDE
When a CMake `add_fuzz_test` target is started from the IDE built in fuzzing configuration, it runs the raw libFuzzer binary.

We should instead let the fuzz test detect that it is not running within cifuzz and let it execute `cifuzz run <fuzz test name>` instead. Coupled with a CMake preset, users can then start fuzzing with cifuzz right from their IDE.

Fixes CLI-337

### PR DESCRIPTION
**Pull request by fmeum**
When built with the fuzzing preset, fuzz test binaries invoke themselves
within `cifuzz run` if started from the command line or IDE.

Fixes #210

### CODE REVIEW

File: tools/cmake/cifuzz/src/launcher.c [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+#define POSIX_EXECLP execlp
+#include <unistd.h>
+#endif
+
+/*
+ * If users launch a fuzz test binary directly, e.g. from the IDE, we want to run cifuzz instead of the raw fuzzer
+ * binary. Since we also want cifuzz and the IDE to share a build directory, we can't implement this distinction at
+ * compile-time - the fuzz test binary must be the same in both cases.
+ *
+ * Instead, we set the CIFUZZ_IS_CIFUZZ environment variable in cifuzz to signal to the fuzz test that it is already
```
[the remark this answers is not in the document]
Comment (jochil) [reply tools/cmake/cifuzz/src/launcher.c:17]: Or if you switch the logic, meaning that the variable tells the launcher to wrap the fuzz target.... `CIFUZZ_INTERCEPT`, `CIFUZZ_WRAP`, `CIFUZZ_INJECT`

**Comment by jochil** [comment]
A reference to the jira task is missing

File: tools/cmake/cifuzz/src/launcher.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (fmeum) [reply tools/cmake/cifuzz/src/launcher.c:17]: I agree that `CIFUZZ_IS_CIFUZZ` reads weird, but I would prefer to keep that variable a generic indicator that some fuzz test should assume that it's running in a cifuzz-equivalent wrapper. 

That the CMake fuzz test launcher does something different if that variable is set isn't relevant for any other part of cifuzz and thus shouldn't be part of the name. I can very well see us tying other behavior to this variable in the future, but cifuzz itself would never need to know about it.

[the remark this answers is not in the document]
Comment (jochil) [reply tools/cmake/cifuzz/src/launcher.c:17]: Ok, makes sense to me.
sorry for being so persistent, but with a weird name like this I would like to give a second thought to it to make it as understandable as possible :see_no_evil: 

> generic indicator that some fuzz test should assume that it's running **_in_** a cifuzz-equivalent wrapper.

so then `CIFUZZ_IN_CIFUZZ`, `CIFUZZ_RUNNING_IN_CIFUZZ`, `CIFUZZ_RUNS_INSIDE`, `CIFUZZ_RUNNING_INSIDE`, `CIFUZZ_RUNS_IN_CIFUZZ`, `CIFUZZ_WRAPPED`?

I would also be open to skip the prefix... I mean CIFUZZ is already present in `IN_CIFUZZ`, `IS_CIFUZZ`

**Comment by fmeum** [comment]
> A reference to the jira task is missing

Added via a GitHub issue.

File: tools/cmake/cifuzz/src/launcher.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (fmeum) [reply tools/cmake/cifuzz/src/launcher.c:17]: > I would also be open to skip the prefix... I mean CIFUZZ is already present in `IN_CIFUZZ`, `IS_CIFUZZ`

We should use a consistent prefix for environment variables that we inject into third-party binaries. It's quite possible that users want to do something differently in their fuzz tests if using cifuzz and they may reasonably choose `IS_CIFUZZ` for that purpose. By having uglier, always-prefixed names, we can minimize the risk of collisions.

**Review by jochil — APPROVED** [review]

File: tools/cmake/cifuzz/src/launcher.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (fmeum) [reply tools/cmake/cifuzz/src/launcher.c]: I like the `IN`, I am using that now instead.

[the remark this answers is not in the document]
Comment (jochil) [reply tools/cmake/cifuzz/src/launcher.c]: Ok. Beside the prefix you would prefer to stay with `CIFUZZ_IS_CIFUZZ`? If yes feel free to close this conversation :)

**Comment by jochil** [comment]
> Added via a GitHub issue.

Maybe we should discuss doing this for all tasks... I would like this (not the extra effort of course)

**Pull request merged**